thorlcr/slave/slavmain.cpp

```diff
@@ -1049,116 +1009,107 @@ class CKJService : public CSimpleInterfaceOf<IKJService>, implements IThreaded,
     CKeyLookupContext *ensureKeyLookupContext(CJobBase &job, const CLookupKey &key, MemoryBuffer &createCtxMb, bool *created=nullptr)
     {
         CriticalBlock b(lCCrit);
-        CKeyLookupContext *keyLookupContext = keyLookupContextsHT.find(key);
-        if (keyLookupContext)
+        auto it = keyLookupContextsHT.find(key);
+        if (it != keyLookupContextsHT.end())
         {
             if (created)
                 *created = false;
-            return LINK(keyLookupContext);
+            return LINK(it->second);
         }
         if (created)
             *created = true;
-        activity_id id = key.id; // JCSMORE annoying, needed because IMPLEMENT_SUPERHASHTABLEOF_REF_FIND doesn't define const (should look at)
-        Linked<CActivityContext> activityCtx = activityContextsHT.find(id);
-        if (!activityCtx)
-        {
-            activityCtx.setown(createActivityContext(job, key.id, createCtxMb));
-            activityContextsHT.replace(*activityCtx); // NB: does not link/take ownership
-        }
-        keyLookupContext = createLookupContext(activityCtx, key);
-        keyLookupContextsHT.replace(*keyLookupContext); // NB: does not link/take ownership
+        Owned<CActivityContext> activityCtx = ensureActivityContext(job, key.id, createCtxMb);
```

Comment:
Note to reviewer: lines 1061-1068 were equivalent to the pre-existing method, ensureActivityContext() - so this is just clearing up.